src/oscar/apps/customer/utils.py

```diff
@@ -23,64 +23,75 @@ def __init__(self, logger=None):
 
     def dispatch_direct_messages(self, recipient, messages):
         """
-        Dispatch one-off messages to explicitly specified recipient(s).
+        Dispatch one-off messages to explicitly specified recipient.
         """
         if messages['subject'] and messages['body']:
-            self.send_email_messages(recipient, messages)
+            return self.send_email_messages(recipient, messages)
 
-    def dispatch_order_messages(self, order, messages, event_type=None,
-                                **kwargs):
+    def dispatch_order_messages(self, order, messages, event_type=None, **kwargs):
         """
-        Dispatch order-related messages to the customer
+        Dispatch order-related messages to the customer.
         """
         if order.is_anonymous:
-            if 'email_address' in kwargs:
-                self.send_email_messages(kwargs['email_address'], messages)
-            elif order.guest_email:
-                self.send_email_messages(order.guest_email, messages)
-            else:
-                return
+            email = kwargs.get('email_address', order.guest_email)
```

Comment:
I'm not totally sure what the question is. The behavior should be the same.

Maybe we need a test for the correct behavior?